### ISSUE TITLE
Will the verification of the author's computer affect the result of the calculation?

### ISSUE THREAD

**Issue by bear-maker** [Warwick-Plasma/epoch#716]
Dear Professor,
I ran epoch on the same computer with the same line of code, but the number of mpiruns was different and the results were different. What is the reason?

**Comment by TomGoffrey** [comment]
What do you mean by "the number of mpiruns was different"? Different number of MPI processes?

Also, when you say the results were different, could you be more specific? Depending on choices of random seed and differing domain decomposition small differences between runs would be expected.

**Comment by bear-maker** [comment]
> What do you mean by "the number of mpiruns was different"? Different number of MPI processes?
> 
> Also, when you say the results were different, could you be more specific? Depending on choices of random seed and differing domain decomposition small differences between runs would be expected.






![1727618250468](https://github.com/user-attachments/assets/526f3105-3a79-4345-a0a0-ed6fe24f96ed)

![1727618250468](https://github.com/user-attachments/assets/08747c46-16f7-47c3-a89f-1f38a3439192)

**Comment by TomGoffrey** [comment]
OK, number of MPI processes. What differences do you observe?

**Comment by bear-maker** [comment]
> OK, number of MPI processes. What differences do you observe?

Thank you for your help. Let me elaborate on this matter in detail.
Comparing the initial electron density of a certain grid, for example, the result I obtained with 80MPI is 2.1e27/cm3, while the result obtained with 40MPI is 2.09e27/cm3. This leads to different results in subsequent calculations. For example, electric and magnetic fields.
A friend told me that I did not enable multi-threaded computation in the code.
